### PR TITLE
Cf 601 inconsistent error code with trailing slash

### PR DESCRIPTION
**Pull request by VinnyQ**

### CODE REVIEW

File: app/src/test/scala/com/rackspace/prefs/FeedsArchivePreferencesTest.scala [hunk 1]
```diff
@@ -53,6 +53,13 @@ class FeedsArchivePreferencesTest extends ScalatraSuite with FunSuiteLike with I
             body should equal (prefs)
             response.mediaType should equal (Some("application/json"))
         }
+
+        info("Calling GET /archive/" + randomId + "/")
```
Comment (shintasmith) [inline app/src/test/scala/com/rackspace/prefs/FeedsArchivePreferencesTest.scala]: This probably should be separated into its own test() { } block, with it's own test message. That way, if one of the GET call failed, we know which one failed by looking at the test message itself.

File: app/src/test/scala/com/rackspace/prefs/FeedsArchivePreferencesTest.scala [hunk 2]
```diff
@@ -88,17 +94,26 @@ class FeedsArchivePreferencesTest extends ScalatraSuite with FunSuiteLike with I
...
         post("/archive/" + randomId, prefs, Map("Content-Type" -> "application/json")) {
             if ( status != 201 ) {
                 info(body)
             }
             status should equal (201)
         }
+
+        val nextRandomId = randomId + 1
+        info("Calling POST /archive/" + nextRandomId + "/")
+        post("/archive/" + nextRandomId + "/", prefs, Map("Content-Type" -> "application/json")) {
```
Comment (shintasmith) [inline app/src/test/scala/com/rackspace/prefs/FeedsArchivePreferencesTest.scala]: Let's move this out to its own separate test() {} block.

File: app/src/test/scala/com/rackspace/prefs/FeedsArchivePreferencesTest.scala [hunk 3]
```diff
@@ -122,51 +138,72 @@ class FeedsArchivePreferencesTest extends ScalatraSuite with FunSuiteLike with I
...
-            """.stripMargin, Map("Content-Type" -> "application/json")) {
+        post("/archive/" + randomId, prefs_enabled, Map("Content-Type" -> "application/json")) {
+            if ( status != 200 ) {
+                info(body)
+            }
+            status should equal (200)
+        }
+
+        info("Calling 3rd POST /archive/" + randomId + "/")
+        post("/archive/" + randomId + "/", prefs_enabled, Map("Content-Type" -> "application/json")) {
```
Comment (shintasmith) [inline app/src/test/scala/com/rackspace/prefs/FeedsArchivePreferencesTest.scala]: This one too... Move to its own separate test() {} block?  Let's discuss. :-)

**Comment by shintasmith** [comment]
Code changes look good... I just have some suggestions on unit test organization.

**Comment by usnavi** [comment]
lgtm other than @shintasmith suggestions.

**Comment by VinnyQ** [comment]
Tests reorganized base on review comments.

**Comment by shintasmith** [comment]
Looks good. Thanks!

File: app/src/main/scala/com/rackspace/prefs/PreferencesService.scala [hunk 4]
```diff
@@ -36,8 +36,9 @@ with JacksonJsonSupport {
         }
     }
 
-    get("/metadata/:preference_slug") {
-        val preferenceSlug = params("preference_slug")
+    get("""^/metadata/([^/?]+)/?""".r) {
+        val uriParts = multiParams("captures")
+        val preferenceSlug = uriParts(0)
```
Comment (ChandraAddala) [inline app/src/main/scala/com/rackspace/prefs/PreferencesService.scala]: Cant something like this work?

http://www.scalatra.org/2.4/guides/http/routes.html#toc_271
```
get("/metadata/:preference_slug/?") {
  //...
}
```

Comment (shintasmith) [inline app/src/main/scala/com/rackspace/prefs/PreferencesService.scala]: @VinnyQ, @ChandraAddala is right. That regex is not necessary. We could just do
```/metadata/:preference_slug/?"

**Pull request merged**